tatanka/client/conn/conn.go

```diff
@@ -178,24 +197,47 @@ type Config struct {
 // TatankaCredentials are the connection credentials for a Tatanka node.
 type TatankaCredentials struct {
 	PeerID tanka.PeerID
-	Addr   string
-	Cert   []byte
-	NoTLS  bool
+	// Addr should not include the protocol prefix.
+	Addr  string
+	Cert  []byte
+	NoTLS bool
+}
+
+func (c *TatankaCredentials) HttpURL() string {
+	if c.NoTLS {
+		return fmt.Sprintf("http://%s", c.Addr)
+	}
+	return fmt.Sprintf("https://%s", c.Addr)
+}
+
+func (c *TatankaCredentials) WsURL() string {
+	if c.NoTLS {
+		return fmt.Sprintf("ws://%s", c.Addr)
+	}
+	return fmt.Sprintf("wss://%s", c.Addr)
```

Comment:
If you want to use sprint here, I think it is advised to escape special characters in the address, which is user supplied correct? https://pkg.go.dev/net/url#QueryEscape

Reply:
`url.QueryEscape` is just for escaping the query string after a URL. We shouldn't have any query strings when connecting to tatanka servers.